### PR TITLE
Implement GetHashCode for SettingSelector

### PR DESCRIPTION
**Pull request by maririos**

### CODE REVIEW

**Comment by maririos** [comment]
**update** : Use the `GetHashBuilder` provided in `Azure.Core`. Thanks @pakrym  for the info

File: src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration/SettingSelector.cs [hunk 1]
```diff
@@ -81,7 +82,16 @@ public override bool Equals(object obj)
...
         [EditorBrowsable(EditorBrowsableState.Never)]
-        public override int GetHashCode() => base.GetHashCode();
+        public override int GetHashCode()
+        {
+            var hashCode = new HashCodeBuilder();
+            hashCode.Add(Keys);
+            hashCode.Add(Labels);
+            hashCode.Add(AsOf);
+            hashCode.Add(Fields);
+            hashCode.Add(BatchLink, StringComparer.InvariantCultureIgnoreCase);
```
Comment (pakrym) [inline src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration/SettingSelector.cs]: Are you sure? Equals uses `StringComparison.Ordinal`  for BatchLink.

Comment (pakrym) [reply src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration/SettingSelector.cs]: Same for ConfigurationSetting.cs

Comment (maririos) [reply src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration/SettingSelector.cs]: My bad :(. 
Fixed

Comment (pakrym) [reply src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration/SettingSelector.cs]: Is there a reason we use `StringComparer.Ordinal`?

**Review by AlexGhiondea — APPROVED** [review]

File: src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration/SettingSelector.cs (hunk 1 above)
Comment (maririos) [reply src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration/SettingSelector.cs]: Talked about this offline.

**Review by pakrym — APPROVED** [review]

**Pull request merged**